SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -34,28 +34,21 @@ def __init__(self, operation, timeout=0):
     def _raise_busy(self):
         raise OperationException('The updater plugin is busy (current operation: {})'.format(self.current_operation))
 
-    def _prelock(self):
-        # 1. Read the current operation from lockfile before timeout or direct lock call.
-        self.current_operation = None
```

Comment:
I forgot this line, we should move it in the `_lock` function now. Just before the `try` to avoid an exception if the field is read. :wink:

Reply:
I saw the line in the catch, but it's probably more safe to put it directly at the start of the function.

Reply:
Yes, there's a case when the attribute is accessed before it is initialized: Filelocker._lock calls _timeout_reached which then calls _raise_busy which reads self.current_operation.